pusher/src/Controller/WokaListController.ts

```diff
@@ -0,0 +1,42 @@
+import { hasToken } from "../Middleware/HasToken";
+import { BaseHttpController } from "./BaseHttpController";
+import { ADMIN_API_URL } from "../Enum/EnvironmentVariable";
+import { adminWokaService } from "..//Services/AdminWokaService";
+import { localWokaService } from "..//Services/LocalWokaService";
+
+export class WokaListController extends BaseHttpController {
+    routes() {
+        // eslint-disable-next-line @typescript-eslint/no-misused-promises
+        this.app.get("/woka-list", { middlewares: [hasToken] }, async (req, res) => {
+            const token = req.header("Authorization");
+            const wokaList = ADMIN_API_URL
+                ? await adminWokaService.getWokaList(token)
+                : await localWokaService.getWokaList(token);
```

Comment:
Even better:

In the constructor, you can do something like:

```
this.wokaService = ADMIN_API_URL ? adminWokaService : localWokaService;
```

Then, anywhere in this controller, you just do:

```ts
const wokaList = this.wokaService.getWokaList();
// ...
const wokaDetails = this.wokaService.fetchWokaDetails();
```